Modules/posixmodule.c

```diff
@@ -13489,8 +13489,10 @@ _Py_COMP_DIAG_POP
     if (self->dir_fd != DEFAULT_DIR_FD) {
 #ifdef HAVE_FSTATAT
       if (HAVE_FSTATAT_RUNTIME) {
+        Py_BEGIN_ALLOW_THREADS
         result = fstatat(self->dir_fd, path, &st,
```

Comment:
`dir_fd` isn't exposed to Python code and it's only set during object creation (in `DirEntry_from_posix_info`) so AFAICT this is safe.